graylog2-server/src/test/java/org/graylog/plugins/views/search/SearchTest.java

```diff
@@ -126,6 +132,44 @@ public void keepsMultipleSearchTypesWhenOverridden() {
         assertThat(searchTypeIdsFrom(after)).containsExactlyInAnyOrder("oans", "gsuffa");
     }
 
+    @Test
+    public void testGetReferencedSearchFiltersIdsReturnsEmptyCollectionOnEmptyQueries() {
+        final Search search = Search.builder().queries(ImmutableSet.of()).build();
+        final Set<String> referencedSearchFiltersIds = search.getReferencedSearchFiltersIds();
+        assertTrue(referencedSearchFiltersIds.isEmpty());
+    }
+
+    @Test
+    public void testGetReferencedSearchFiltersIdsReturnsEmptyCollectionOnQueriesWithoutFilters() {
+        final Query query1 = mock(Query.class);
```

Comment:
There are at least 3 required fields that I'd need to provide while building the query with builder, so it would be a long builder invocation... And the only thing I really care about Query are filters...

Reply:
I know that it adds some code. On the other hand you'll get a proper fully initialized valid query instance, not just a mock. I think this is worth the additional effort. You could wrap the builder into a method call to minimize the change in the test itself. 

There are certainly valid use cases for Mockito, but I don't think we need to go as far as mocking our own DTOs :-)

Reply:
I could even use validQueryBuilder() from TestData class if you think it is worth the effort. The one that was created to avoid mocks, but has some mocking inside anyway... ;)